File: INTRODUCCION A POO/EJERCICIO-1/Persona.py
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad

    def saludar(self):
        print(f"Hola, soy {self.nombre} de {self.ciudad}")

    def es_mayor_de_edad(self):
        return self.edad >= 18


persona1 = Persona("Juan", 25, "Tarija")
persona2 = Persona("Ana", 17, "Cochabamba")
persona3 = Persona("Carlos", 30, "La Paz")

persona1.saludar()
persona2.saludar()
persona3.saludar()

print(f"\n¿{persona1.nombre} es mayor de edad? {persona1.es_mayor_de_edad()}")
print(f"¿{persona2.nombre} es mayor de edad? {persona2.es_mayor_de_edad()}")
print(f"¿{persona3.nombre} es mayor de edad? {persona3.es_mayor_de_edad()}")

File: INTRODUCCION A POO/EJERCICIO-1/Persona/src/ejercicio1/Persona.java
package ejercicio1;

public class Persona {
    private String nombre;
    private int edad;
    private String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    public void saludar() {
        System.out.println("Hola, soy " + this.nombre + " de " + this.ciudad);
    }

    public boolean esMayorDeEdad() {
        return this.edad >= 18;
    }

    public static void main(String[] args) {
        Persona persona1 = new Persona("Juan", 25, "Tarija");
        Persona persona2 = new Persona("Ana", 17, "Cochabamba");
        Persona persona3 = new Persona("Carlos", 30, "La Paz");

        persona1.saludar();
        persona2.saludar();
        persona3.saludar();

        System.out.println("\n¿" + persona1.nombre + " es mayor de edad? " + persona1.esMayorDeEdad());
        System.out.println("¿" + persona2.nombre + " es mayor de edad? " + persona2.esMayorDeEdad());
        System.out.println("¿" + persona3.nombre + " es mayor de edad? " + persona3.esMayorDeEdad());
    }
}